test/libsolidity/SemanticTest.cpp

```diff
@@ -88,9 +88,15 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
 	if (m_runWithEwasm && !m_supportsEwasm)
 		m_runWithEwasm = false;
 
-	m_runWithABIEncoderV1Only = m_reader.boolSetting("ABIEncoderV1Only", false);
-	if (m_runWithABIEncoderV1Only && !solidity::test::CommonOptions::get().useABIEncoderV1)
-		m_shouldRun = false;
+	if (!solidity::test::CommonOptions::get().useABIEncoderV1)
+	{
+		for (auto const& [filename,source]: m_reader.sources().sources)
+			if (source.find("pragma abicoder v1;") != string::npos)
```

Comment:
> pragma abicoder "v1" from #10795.

I would rather argue while that is accepted, it is not something we should promote in the tests. So okay to leave it out.

> pragma abicoder v1 hack used by @chriseth to avoid changing source locations.

What do you mean?

Reply:
> What do you mean?

See https://github.com/ethereum/solidity/pull/10351#discussion_r526992754

> I would rather argue while that is accepted, it is not something we should promote in the tests. So okay to leave it out.

But this way a test can get silently skipped just because someone used quotes or inserted an extra space. It would be fine if there was an error on weirdly formatted pragmas in tests but I don't think that's the case.

A more flexible solution is easy - just use a regex.

Reply:
If it is not detected by this simple check then tests will fail, so the PR won't be merged. I'd argue we don't need to make this bulletproof as we want some consistency in these tests. I could change it to a regexp, but really don't see a huge benefit in doing so.

Reply:
And as for ensuring there are no source location changes, again I do not fully agree with that approach.

Reply:
Well, ok, after looking at it again I think you're right. The way this is written now a failure to match a weirdly formatted pragma means just that the test will **not** be skipped. Might be confusing but yeah, won't silently hide anything.